lib/OpenQA/WebAPI/Controller/Admin/User.pm

```diff
@@ -39,17 +43,45 @@ sub update {
         $is_operator = 1;
     }
 
+    my $data = {is_admin => $is_admin, is_operator => $is_operator};
+    $data->{username} = $username if (defined($username));
```

Comment:
The `username` column is used as an ID when using OpenID and possibly other providers. In case of OpenID it is an URL like `https://www.opensuse.org/openid/user/someuser` and it is very questionable whether we should allow to change that. It would basically allow migrating an "openQA user" to a different underlying OpenID user.